File: backend/src/models/Bodega.ts
import RegionVitivinicola from './RegionVitivinicola'

export default class Bodega {
  private nombre: string
  private region: RegionVitivinicola
  constructor(nombre: string, region: RegionVitivinicola) {
    this.nombre = nombre
    this.region = region
  }

  obtenerNombreYRegion() {
    return [this.getNombre(), [this.region.getNombre(), this.getRegion()]]
  }

  getNombre(): string {
    return this.nombre
  }

  getRegion(): RegionVitivinicola {
    return this.region
  }
}

File: ReporteRankingVinos/src/main/java/com/Bonvino/ReporteRankingVinos/models/Bodega.java
package com.Bonvino.ReporteRankingVinos.models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
public class Bodega {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  private String nombre;

  @ManyToOne
  @JoinColumn(name = "regionId", nullable = false)
  private RegionVitivinicola region;

  public String obtenerNombreRegionProvinciaYPais() {
    return this.getNombre() + ", " + this.region.getNombreProvinciaYPais();
  }

  public Bodega(String nombre, RegionVitivinicola region) {
    this.nombre = nombre;
    this.region = region;
  }
}
